server/multitenancy/tenant_resolver.ts

```diff
@@ -194,3 +199,32 @@ export function resolve(
 export function isValidTenant(tenant: string | undefined | null): boolean {
   return tenant !== undefined && tenant !== null;
 }
+
+/**
+ * If multitenancy is enabled & the URL entered starts with /goto,
+ * We will modify the rawResponse to add a new parameter to the URL, the security_tenant (or value for tenant when in multitenancy)
+ * With the security_tenant added, the resolved short URL now contains the security_tenant information (so the short URL retains the tenant information).
+ **/
+
+export function addTenantParameterToResolvedShortLink(request: OpenSearchDashboardsRequest) {
+  if (request.url.pathname.startsWith(`${GOTO_PREFIX}/`)) {
+    const rawRequest = ensureRawRequest(request);
+    const rawResponse = rawRequest.response as ResponseObject;
+
+    // Make sure the request really should redirect
+    if (rawResponse.headers.location) {
+      const modifiedUrl = modifyUrl(rawResponse.headers.location as string, (parts) => {
+        if (typeof parts.query.security_tenant === 'undefined') {
+          parts.query.security_tenant = request.headers.securitytenant as string;
+        }
+      });
+
+      // Mutating the headers toolkit.next({headers: ...}) logs a warning about headers being overwritten
+      if (rawResponse.headers.location) {
```

Comment:
why check for this condition?

Reply:
Thanks Darshit I see now there is no need for it as it's checked further up in the function.

I've amended the code.